pkg/apis/streaming/v1alpha1/processor_validation.go

```diff
@@ -132,6 +133,21 @@ func (s *ProcessorSpec) validateStreamAliasUniqueness() validation.FieldErrors {
 		}
 		uses[alias] = append(uses[alias], fmt.Sprintf("inputs[%d].alias", i))
 	}
+	for _, alias := range aliases {
+		if len(uses[alias]) > 1 {
+			errs = errs.Also(validation.ErrDuplicateValue(alias, uses[alias]...))
+		}
+	}
+
+	return errs
```

Comment:
you could still extract common code between Inputs and Outputs, as the validation is the same:
```go
func validateAliasUniqueness(allAliases []string, aliasType string) validation.FieldErrors {
	errs := validation.FieldErrors{}

	var dedupedAliases []string
	uses := map[string][]string{}
	for i, alias := range allAliases {
		if _, ok := uses[alias]; !ok {
			uses[alias] = []string{}
			dedupedAliases = append(dedupedAliases, alias)
		}
		uses[alias] = append(uses[alias], fmt.Sprintf("%s[%d].alias", aliasType, i))
	}

	for _, alias := range dedupedAliases {
		if len(uses[alias]) > 1 {
			errs = errs.Also(validation.ErrDuplicateValue(alias, uses[alias]...))
		}
	}
	return errs
}
```
then, the calling code becomes:
```go
func (s *ProcessorSpec) validateStreamInputAliasUniqueness() validation.FieldErrors {
	var aliases []string
	for _, input := range s.Inputs {
		aliases = append(aliases, input.Alias)
	}
	return validateAliasUniqueness(aliases, "inputs")
}
// [...]
func (s *ProcessorSpec) validateStreamOutputAliasUniqueness() validation.FieldErrors {
	var aliases []string
	for _, output := range s.Outputs {
		aliases = append(aliases, output.Alias)
	}
	return validateAliasUniqueness(aliases, "outputs")
}
```